awx/ui_next/src/setupCSP.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable */
+
+// Set a special variable to add `nonce` attributes to all styles/script tags
+// See https://github.com/webpack/webpack/pull/3210
+__webpack_nonce__ = window.NONCE_ID;
+
+// Send report when a CSP violation occurs
+// See: https://w3c.github.io/webappsec-csp/2/#violation-reports
+// See: https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent
+document.addEventListener('securitypolicyviolation', e => {
+  const violation = {
+    'csp-report': {
+      'blocked-uri': e.blockedURI,
+      'document-uri': e.documentURI,
+      'effective-directive': e.effectiveDirective,
+      'original-policy': e.originalPolicy,
+      referrer: e.referrer,
+      'status-code': e.statusCode,
+      'violated-directive': e.violatedDirective,
+    },
+  };
+  if (e.sourceFile) violation['csp-report']['source-file'] = e.sourceFile;
+  if (e.lineNumber) violation['csp-report']['line-number'] = e.lineNumber;
+  if (e.columnNumber) violation['csp-report']['column-number'] = e.columnNumber;
+
+  const xhr = new XMLHttpRequest();
+  xhr.open('POST', '/csp-violation/', true);
+  xhr.setRequestHeader('content-type', 'application/csp-report');
```

Comment:
This may have to be `application/json` for DRF to be happy (and not require a bunch of weird gymnastics).

Reply:
The specification is to use a `Content-Type` header field of `application/csp-report`: https://w3c.github.io/webappsec-csp/2/#violation-reports

Reply:
@nixocio Tells me this is already wired up: https://github.com/ansible/awx/blob/devel/awx/urls.py#L25

![Screenshot from 2020-12-07 20-52-47](https://user-images.githubusercontent.com/9753817/101428233-4f501980-38ce-11eb-82f5-a78b72766495.png)

:man_shrugging:

Reply:
https://github.com/ansible/awx/commit/eacf819caf4

https://images.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.giphy.com%2Fmedia%2FFPjbHO0jJxGsE%2Fgiphy.gif&f=1